### PR TITLE
Rework the exit procedure a little bit

### PR DESCRIPTION
**Pull request by sh1r4s3**

### CODE REVIEW

File: lispy.c [hunk 1]
```diff
@@ -1439,13 +1446,17 @@ int main(int argc, char** argv){
     lenv_add_builtins(e);
     lval* standard = lval_add(lval_sexpr(), lval_str("std-library/std_library.lspy"));
     lval* std = builtin_load(e, standard);
+    if (atexit(cleanup)) {
```
Comment (Federico-abss) [inline lispy.c:1449]: doesn't this trigger the conditional when atexit is called correctly? It should be called in case of error

**Review by Federico-abss — COMMENTED** [review]
Can you let me know how about line 1449

**Review by Federico-abss — APPROVED** [review]
Tried the change and works great

File: lispy.c (hunk 1 above)
Comment (Federico-abss) [reply lispy.c:1449]: Actually tried the change and works correctly, could you still explain to me how it does exactly work and how to trigger the error message?
Really grateful for your help, you are really knowledgeable

Comment (sh1r4s3) [reply lispy.c:1449]: According to the atexit(3) [1] the atexit function would return zero in the case of success and non zero in the case of error.

[1] http://man7.org/linux/man-pages/man3/atexit.3.html

Comment (Federico-abss) [reply lispy.c:1449]: Ok I read the opposite thing in a forum and clearly was wrong, thanks for the resources

Comment (sh1r4s3) [reply lispy.c:1449]: Your welcome :)

**Pull request merged**